src/Microsoft.Framework.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -9,23 +9,8 @@ public interface IConfiguration
     {
         string this[string key] { get; set; }
 
-        /// <summary>
-        /// 
-        /// </summary>
-        /// <param name="key">A case insensitive name.</param>
-        /// <returns>The value associated with the given key, or null if none is found.</returns>
-        string Get(string key);
+        IConfigurationSection GetSection(string key);
 
-        bool TryGet(string key, out string value);
-
-        IConfiguration GetConfigurationSection(string key);
-
-        IEnumerable<KeyValuePair<string, IConfiguration>> GetConfigurationSections();
-
-        IEnumerable<KeyValuePair<string, IConfiguration>> GetConfigurationSections(string key);
-
-        void Set(string key, string value);
-
-        void Reload();
+        IEnumerable<IConfigurationSection> GetChildren();
```

Comment:
`GetChildren()` returns the direct children of the current node, while `GetSection(key)` returns a sección addressed by key relativa to the current node. The latter can be a direct child or a decendant several levels below. At some point i was convinced that the best name for `GetChildren()` was `GetChildSections()` but after discussing it a lot with @lodejard and the BCL folks we agreed on what we have and it has grown on me.